x-pack/plugins/actions/server/actions_client.ts

```diff
@@ -0,0 +1,174 @@
+  description: string;
+  actionTypeId: string;
+  actionTypeConfig: Record<string, any>;
+}
+
+interface FireOptions {
+  id: string;
+  params: Record<string, any>;
+}
+
+interface CreateOptions {
+  data: Action;
+  options?: {
+    migrationVersion?: Record<string, string>;
+    references?: SavedObjectReference[];
+  };
+}
+
+interface FindOptions {
+  options?: {
+    perPage?: number;
+    page?: number;
+    search?: string;
+    defaultSearchOperator?: 'AND' | 'OR';
+    searchFields?: string[];
+    sortField?: string;
+    hasReference?: {
+      type: string;
+      id: string;
+    };
+    fields?: string[];
+  };
+}
+
+interface ConstructorOptions {
+  actionTypeService: ActionTypeService;
+  encryptedSavedObjectsPlugin: EncryptedSavedObjectsPlugin;
+  savedObjectsClient: SavedObjectsClient;
+}
+
+interface UpdateOptions {
+  id: string;
+  data: Action;
+  options: { version?: string; references?: SavedObjectReference[] };
+}
+
+export class ActionsClient {
+  private savedObjectsClient: SavedObjectsClient;
+  private actionTypeService: ActionTypeService;
+  private encryptedSavedObjectsPlugin: EncryptedSavedObjectsPlugin;
+
+  constructor({
+    actionTypeService,
+    encryptedSavedObjectsPlugin,
+    savedObjectsClient,
+  }: ConstructorOptions) {
+    this.actionTypeService = actionTypeService;
+    this.encryptedSavedObjectsPlugin = encryptedSavedObjectsPlugin;
+    this.savedObjectsClient = savedObjectsClient;
+  }
+
+  /**
+   * Create an action
+   */
+  public async create({ data, options }: CreateOptions) {
+    const { actionTypeId } = data;
+    if (!this.actionTypeService.has(actionTypeId)) {
+      throw Boom.badRequest(
+        i18n.translate('xpack.actions.actionsClient.create.missingActionTypeError', {
+          defaultMessage: 'Action type "{actionTypeId}" is not registered.',
+          values: {
+            actionTypeId,
+          },
+        })
+      );
+    }
+    this.actionTypeService.validateActionTypeConfig(actionTypeId, data.actionTypeConfig);
+    const actionWithSplitActionTypeConfig = this.moveEncryptedAttributesToSecrets(data);
+    return await this.savedObjectsClient.create('action', actionWithSplitActionTypeConfig, options);
+  }
+
+  /**
+   * Get an action
+   */
+  public async get({ id }: { id: string }) {
+    return await this.savedObjectsClient.get('action', id);
+  }
+
+  /**
+   * Find actions
+   */
+  public async find({ options = {} }: FindOptions) {
+    return await this.savedObjectsClient.find({
+      ...options,
+      type: 'action',
+    });
+  }
+
+  /**
+   * Delete action
+   */
+  public async delete({ id }: { id: string }) {
+    return await this.savedObjectsClient.delete('action', id);
+  }
+
+  /**
+   * Update action
+   */
+  public async update({ id, data, options = {} }: UpdateOptions) {
+    const { actionTypeId } = data;
+    if (!this.actionTypeService.has(actionTypeId)) {
+      throw Boom.badRequest(`Action type "${actionTypeId}" is not registered.`);
```

Comment:
Yeah for this one I think changing it to `this.actionTypeService.get(actionTypeId);` would do the trick and let it throw an error. In this function here, it's a bit tricky since we do support partial updates where the `if (data.actionTypeConfig) {` can be skipped.